Failed Robolectric 4 unit tests running with Gradle

Hello,
### Description
I faced a problem when I tried to run the Robolectric unit tests with gradle. The error I got is the following, not descriptive log:

```
MyActivityTest > shouldNotBeNull FAILED
    java.lang.NullPointerException
MyActivityTest: sdk=28; resources=binary
java.lang.RuntimeException
        Caused by: android.content.pm.PackageParser$PackageParserException
            Caused by: java.lang.NullPointerException
```

### Steps to Reproduce
```groovy
testOptions {
        unitTests {
            //returnDefaultValues = true
            includeAndroidResources = true
        }
    }
```
Add the following line to `gradle.properties`:
```properties
android.enableUnitTestBinaryResources=true
```

```java
import org.junit.Before;
import org.junit.Rule;
import org.junit.Test;
import org.junit.runner.RunWith;

import androidx.test.ext.junit.runners.AndroidJUnit4;
import androidx.test.rule.ActivityTestRule;

import static androidx.test.espresso.Espresso.onView;
import static androidx.test.espresso.action.ViewActions.click;
import static androidx.test.espresso.assertion.ViewAssertions.matches;
import static androidx.test.espresso.matcher.ViewMatchers.isEnabled;
import static androidx.test.espresso.matcher.ViewMatchers.withId;
import static androidx.test.espresso.matcher.ViewMatchers.withText;
import static org.junit.Assert.assertNotNull;

@RunWith(AndroidJUnit4.class)
public class MyActivityTest {


    private MyActivity activity;

    @Rule
    public ActivityTestRule< MyActivity > mActivityRule = new ActivityTestRule<>(MyActivity.class);

    @Before
    public void setUp() {
        activity = mActivityRule.getActivity();
    }

    @Test
    public void shouldNotBeNull() throws Exception {
        assertNotNull( activity );
    }
```
Run the tests with:
`./gradlew testProductionDebugUnitTest`

### Robolectric & Android Version
Robolectric 4 beta 1 - Android 3.2.1 and Gradle plugin 3.2.1

I got the same type of error : 

```
java.lang.RuntimeException: android.content.pm.PackageParser$PackageParserException: Failed to read manifest from /Users/hugo/dev/java/beschool-android/app/build/intermediates/apk_for_local_test/autoTestDebugUnitTest/packageAutoTestDebugUnitTestForUnitTest/apk-for-local-test.ap_

	at org.robolectric.shadows.ShadowPackageParser.callParsePackage(ShadowPackageParser.java:57)
	at org.robolectric.android.internal.ParallelUniverse.setUpApplicationState(ParallelUniverse.java:142)
	at org.robolectric.RobolectricTestRunner.beforeTest(RobolectricTestRunner.java:372)
	at org.robolectric.internal.SandboxTestRunner$2.evaluate(SandboxTestRunner.java:252)
	at org.robolectric.internal.SandboxTestRunner.runChild(SandboxTestRunner.java:130)
	at org.robolectric.internal.SandboxTestRunner.runChild(SandboxTestRunner.java:42)
	at org.junit.runners.ParentRunner$3.run(ParentRunner.java:290)
	at org.junit.runners.ParentRunner$1.schedule(ParentRunner.java:71)
	at org.junit.runners.ParentRunner.runChildren(ParentRunner.java:288)
	at org.junit.runners.ParentRunner.access$000(ParentRunner.java:58)
	at org.junit.runners.ParentRunner$2.evaluate(ParentRunner.java:268)
	at org.robolectric.internal.SandboxTestRunner$1.evaluate(SandboxTestRunner.java:84)
	at org.junit.runners.ParentRunner.run(ParentRunner.java:363)
	at org.junit.runner.JUnitCore.run(JUnitCore.java:137)
	at com.intellij.junit4.JUnit4IdeaTestRunner.startRunnerWithArgs(JUnit4IdeaTestRunner.java:68)
	at com.intellij.rt.execution.junit.IdeaTestRunner$Repeater.startRunnerWithArgs(IdeaTestRunner.java:47)
	at com.intellij.rt.execution.junit.JUnitStarter.prepareStreamsAndStart(JUnitStarter.java:242)
	at com.intellij.rt.execution.junit.JUnitStarter.main(JUnitStarter.java:70)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at com.intellij.rt.execution.application.AppMainV2.main(AppMainV2.java:131)
Caused by: android.content.pm.PackageParser$PackageParserException: Failed to read manifest from /Users/hugo/dev/java/beschool-android/app/build/intermediates/apk_for_local_test/autoTestDebugUnitTest/packageAutoTestDebugUnitTestForUnitTest/apk-for-local-test.ap_
	at android.content.pm.PackageParser.parseBaseApk(PackageParser.java:1353)
	at android.content.pm.PackageParser.parseMonolithicPackage(PackageParser.java:1299)
	at android.content.pm.PackageParser.parsePackage(PackageParser.java:1022)
	at android.content.pm.PackageParser.parsePackage(PackageParser.java:1042)
	at org.robolectric.shadows.ShadowPackageParser.callParsePackage(ShadowPackageParser.java:31)
	at org.robolectric.android.internal.ParallelUniverse.setUpApplicationState(ParallelUniverse.java:142)
	at org.robolectric.RobolectricTestRunner.beforeTest(RobolectricTestRunner.java:372)
	at org.robolectric.internal.SandboxTestRunner$2.evaluate(SandboxTestRunner.java:252)
	at org.robolectric.internal.SandboxTestRunner.runChild(SandboxTestRunner.java:130)
	at org.robolectric.internal.SandboxTestRunner.runChild(SandboxTestRunner.java:42)
	at org.junit.runners.ParentRunner$3.run(ParentRunner.java:290)
	at org.junit.runners.ParentRunner$1.schedule(ParentRunner.java:71)
	at org.junit.runners.ParentRunner.runChildren(ParentRunner.java:288)
	at org.junit.runners.ParentRunner.access$000(ParentRunner.java:58)
	at org.junit.runners.ParentRunner$2.evaluate(ParentRunner.java:268)
	at org.robolectric.internal.SandboxTestRunner$1.evaluate(SandboxTestRunner.java:84)
	at org.junit.runners.ParentRunner.run(ParentRunner.java:363)
	at org.junit.runner.JUnitCore.run(JUnitCore.java:137)
	at com.intellij.junit4.JUnit4IdeaTestRunner.startRunnerWithArgs(JUnit4IdeaTestRunner.java:68)
	at com.intellij.rt.execution.junit.IdeaTestRunner$Repeater.startRunnerWithArgs(IdeaTestRunner.java:47)
	at com.intellij.rt.execution.junit.JUnitStarter.prepareStreamsAndStart(JUnitStarter.java:242)
	at com.intellij.rt.execution.junit.JUnitStarter.main(JUnitStarter.java:70)
	... 1 more
Caused by: java.lang.NullPointerException
	at org.robolectric.res.android.CppAssetManager2.FindEntry(CppAssetManager2.java:678)
	at org.robolectric.res.android.CppAssetManager2.GetResource(CppAssetManager2.java:841)
	at org.robolectric.res.android.CppAssetManager2.ResolveReference(CppAssetManager2.java:899)
	at org.robolectric.res.android.AttributeResolution9.RetrieveAttributes(AttributeResolution9.java:484)
	at org.robolectric.shadows.ShadowArscAssetManager9.nativeRetrieveAttributes(ShadowArscAssetManager9.java:1454)
	at android.content.res.AssetManager.nativeRetrieveAttributes(AssetManager.java)
	at android.content.res.AssetManager.retrieveAttributes(AssetManager.java:1012)
	at android.content.res.Resources.obtainAttributes(Resources.java:1814)
	at android.content.res.Resources$GeneratedProxy/361380654.obtainAttributes(Unknown Source)
	at org.robolectric.shadows.ShadowResources.obtainAttributes(ShadowResources.java:90)
	at android.content.res.Resources.obtainAttributes(Resources.java)
	at android.content.pm.PackageParser.parseMetaData(PackageParser.java:5318)
	at android.content.pm.PackageParser.parseProviderTags(PackageParser.java:4961)
	at android.content.pm.PackageParser.parseProvider(PackageParser.java:4927)
	at android.content.pm.PackageParser.parseBaseApplication(PackageParser.java:3671)
	at android.content.pm.PackageParser.parseBaseApkCommon(PackageParser.java:2053)
	at android.content.pm.PackageParser.parseBaseApk(PackageParser.java:1942)
	at android.content.pm.PackageParser.parseBaseApk(PackageParser.java:1337)
	... 22 more
```

--- a/resources/src/main/java/org/robolectric/res/android/ByteBucketArray.java
+++ b/resources/src/main/java/org/robolectric/res/android/ByteBucketArray.java
@@ -6,6 +6,10 @@ package org.robolectric.res.android;
  * (number of entries that a byte can represent).
  */
 public abstract class ByteBucketArray<T> {
+  public ByteBucketArray(T mDefault) {
+    this.mDefault = mDefault;
+  }
+
   final int size() {
     return NUM_BUCKETS * BUCKET_SIZE;
   }
@@ -25,7 +29,8 @@ public abstract class ByteBucketArray<T> {
     if (bucket == null) {
       return mDefault;
     }
-    return bucket[0x0f & ((byte) index)];
+    T t = bucket[0x0f & ((byte) index)];
+    return t == null ? mDefault : t;
   }
 
   T editItemAt(int index) {

--- a/resources/src/main/java/org/robolectric/res/android/CppAssetManager2.java
+++ b/resources/src/main/java/org/robolectric/res/android/CppAssetManager2.java
@@ -1279,12 +1279,13 @@ public class CppAssetManager2 {
         //
         // // Re-create it.
         // new (impl.filtered_configs_) ByteBucketArray<FilteredConfigGroup>();
-        impl.filtered_configs_ = new ByteBucketArray<FilteredConfigGroup>() {
-          @Override
-          FilteredConfigGroup newInstance() {
-            return new FilteredConfigGroup();
-          }
-        };
+        impl.filtered_configs_ =
+            new ByteBucketArray<FilteredConfigGroup>(new FilteredConfigGroup()) {
+              @Override
+              FilteredConfigGroup newInstance() {
+                return new FilteredConfigGroup();
+              }
+            };
 
         // Create the filters here.
         impl.loaded_package_.ForEachTypeSpec((TypeSpec spec, byte type_index) -> {

--- a/resources/src/main/java/org/robolectric/res/android/ResTable.java
+++ b/resources/src/main/java/org/robolectric/res/android/ResTable.java
@@ -2570,12 +2570,13 @@ public class ResTable {
     // Gets cleared whenever the parameters/configuration changes.
     // These are stored here in a parallel structure because the data in `types` may
     // be shared by other ResTable's (framework resources are shared this way).
-    ByteBucketArray<TypeCacheEntry> typeCacheEntries = new ByteBucketArray<TypeCacheEntry>() {
-      @Override
-      TypeCacheEntry newInstance() {
-        return new TypeCacheEntry();
-      }
-    };
+    ByteBucketArray<TypeCacheEntry> typeCacheEntries =
+        new ByteBucketArray<TypeCacheEntry>(new TypeCacheEntry()) {
+          @Override
+          TypeCacheEntry newInstance() {
+            return new TypeCacheEntry();
+          }
+        };
 
     // The table mapping dynamic references to resolved references for
     // this package group.

